x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtIssuerHttpsServer.java

```diff
@@ -0,0 +1,98 @@
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.nio.file.Path;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+
+/**
+ * HTTPS server for JWT issuer to host a public PKC JWKSet.
+ */
+public class JwtIssuerHttpsServer implements Closeable {
+    private static final Logger LOGGER = LogManager.getLogger(JwtIssuerHttpsServer.class);
+
+    private static final String ADDRESS = "localhost"; // localhost, 127.0.0.1, ::1, hostname, FQDN, etc
+    private static final int PORT = 0; // 443, 0 (ephemeral port)
+    private static final int BACKLOG = 0; // max queued incoming connections
+    private static final int STOP_DELAY_SECONDS = 0; // 0 no limit, >0 limited
+    private static final String PATH = "/valid/"; // Tests can call other paths like "/invalid/" to verify graceful HTTP 404 error handling
+
+    static final Path CERT_PATH = JwtTestCase.resolvePath("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.crt");
+    private static final Path KEY_PATH = JwtTestCase.resolvePath(
+        "/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.pem"
+    );
+    private static final char[] PASSWORD = "testnode".toCharArray();
+
+    private final HttpsServer httpsServer;
+    final String url; // JWT realm needs this for HTTP GET requests
+
+    @SuppressForbidden(reason = "MockHttpServer.createHttps requires InetSocketAddress, PORT=0 resolves to an available ephemeral port.")
+    public JwtIssuerHttpsServer(final byte[] encodedJwkSetPkcPublicBytes) throws Exception {
+        this.httpsServer = MockHttpServer.createHttps(new InetSocketAddress(ADDRESS, PORT), BACKLOG);
+        this.url = "https://" + ADDRESS + ":" + this.httpsServer.getAddress().getPort() + PATH; // get ephemeral port
+        this.httpsServer.setHttpsConfigurator(new HttpsConfigurator(this.createSslContext()));
+        this.httpsServer.createContext("/", new JwtIssuerHttpHandler(encodedJwkSetPkcPublicBytes));
+        LOGGER.trace("Starting [{}]", this.url);
+        this.httpsServer.start();
+        LOGGER.debug("Started [{}]", this.url);
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (this.httpsServer != null) {
+            LOGGER.trace("Stopping [{}]", this.url);
+            this.httpsServer.stop(STOP_DELAY_SECONDS);
+            LOGGER.debug("Stopped [{}]", this.url);
+        }
+    }
+
+    private SSLContext createSslContext() throws Exception {
+        final String tlsProtocol = ESTestCase.inFipsJvm() ? "TLSv1.2" : ESTestCase.randomFrom("TLSv1.2", "TLSv1.3");
+        final SSLContext sslContext = SSLContext.getInstance(tlsProtocol);
+        final KeyManager keyManager = CertParsingUtils.getKeyManagerFromPEM(CERT_PATH, KEY_PATH, PASSWORD);
+        sslContext.init(new KeyManager[] { keyManager }, null, JwtTestCase.SECURE_RANDOM);
```

Comment:
Nit: It does not seem necesary to have an constant (`SECURE_RANDOM`) defined in `JwtTestCase` so it can be passed in here. This argument can as well be `null` or inlined.